### PR TITLE
Implement emscripten_async_load_script under node

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: src/library_browser.js [hunk 1]
```diff
@@ -866,7 +888,7 @@ var LibraryBrowser = {
       {{{ runtimeKeepalivePop() }}}
       if (onerror) onerror();
     };
-    script.src = UTF8ToString(url);
+    script.src = url
```
Comment (kripken) [inline src/library_browser.js:891]:
```suggestion
    script.src = url;
```

File: src/library_browser.js [hunk 2]
```diff
@@ -832,27 +832,49 @@ var LibraryBrowser = {
...
     }
 #endif
+    assert(runDependencies === 0, 'async_load_script must be run when no other dependencies are active');
+
     {{{ runtimeKeepalivePush() }}}
 
-    assert(runDependencies === 0, 'async_load_script must be run when no other dependencies are active');
+#if ENVIRONMENT_MAY_BE_NODE
+    if (ENVIRONMENT_IS_NODE) {
+      readAsync(url, (data) => {
```
Comment (kripken) [inline src/library_browser.js:857]: It looks like the functions here can be shared with below? The main difference is just in calling `readAsync` vs setting them on `.onload/onerror`.

**Comment by sbc100** [comment]
As a followup we can use the same implementation to remove the current main-thread-only limitation of this function.

File: src/library_browser.js [hunk 3]
```diff
@@ -861,12 +861,27 @@ var LibraryBrowser = {
...
+      }, loadError, false);
+      return;
+    }
+#endif
+
+    var script = document.createElement('script');
+    script.onload = loadDone;
+    script.onerror = loadError;
+    script.src = url;
     document.body.appendChild(script);
```
Comment (kripken) [inline src/library_browser.js:885]: I wonder if there could be a function that abstracts over both of these code paths, but if this is the only place we have them then maybe it's fine as it is.

**Review by kripken — APPROVED** [review]

**Pull request merged**